graylog2-server/src/main/java/org/graylog2/indexer/searches/Searches.java

```diff
@@ -263,11 +263,19 @@ public SearchResult search(SearchesConfig config) {
         return new SearchResult(r.getHits(), indices, config.query(), request.source(), r.getTook());
     }
 
-    public TermsResult terms(String field, int size, String query, String filter, TimeRange range) {
+    public TermsResult terms(String field, int size, String query, String filter, TimeRange range, Sorting.Direction sorting) {
+        Terms.Order termsOrder;
+
         if (size == 0) {
             size = 50;
         }
 
+        if (sorting == Sorting.Direction.DESC){
+            termsOrder = Terms.Order.count(false);
```

Comment:
Hi @joschi. I am using count because I am ordering basing it on the returned document count of the aggregation instead of just the term itself.

Per the docs: https://www.elastic.co/guide/en/elasticsearch/client/java-api/2.4/_bucket_aggregations.html#_order
```
Ordering the buckets by their doc_count in an ascending manner:

AggregationBuilders
    .terms("genders")
    .field("gender")
    .order(Terms.Order.count(true))
```